File: 2017 - 2/Programação de Computadores/All Netbeans Projects/NetBeansProjects/C++/ObraDeArte.cpp
#include "ObraDeArte.h"

namespace pc {

    ObraDeArte::ObraDeArte(string titulo, string artista, string material, int ano) :
    titulo(titulo), artista(artista), material(material), ano(ano) {
    }

    ObraDeArte::~ObraDeArte() {
    }

    int ObraDeArte::getAno() const {
        return ano;
    }

    string ObraDeArte::getArtista() const {
        return artista;
    }

    string ObraDeArte::getMaterial() const {
        return material;
    }

    string ObraDeArte::getTitulo() const {
        return titulo;
    }

}

File: 2017 - 2/Programação de Computadores/projetos Java prova Delano/Polimorfismo Parametrico/Polipara e abstract classes (1)/Java/src/br/ufscar/dc/pc/ObraDeArte.java
package br.ufscar.dc.pc;

/**
 * Classe ObraDeArte
 *
 * @author Delano Medeiros Beder
 */
public abstract class ObraDeArte {
    
    /*
     * Declaração dos atributos da classe
     */
    private String título, artista, material;
    
    private int ano;

    /*
     * Declaração do construtor da classe
     */
    
    public ObraDeArte(String título, String artista, String material, int ano) {
        this.título = título;
        this.artista = artista;
        this.material = material;
        this.ano = ano;
    }

    /*
     * Declaração dos métodos da classe
     */
    
    public abstract void imprime();
    
    public String getTítulo() {
        return título;
    }

    public String getArtista() {
        return artista;
    }

    public String getMaterial() {
        return material;
    }

    public int getAno() {
        return ano;
    }
}
